src/avram/associations/has_many.cr

```diff
@@ -85,13 +102,10 @@ module Avram::Associations::HasMany
 
     def {{ assoc_name.id }}_count : Int64
       {% if through %}
-        {{ model }}::BaseQuery
-          .new
-          .join_{{ through.id }}
-          .__yield_where_{{ through.id }} do |through_query|
-            through_query.{{ foreign_key.id }}(id)
-          end
-          .select_count
+        {{ through.first.id }}_query
+          .{{ foreign_key.id }}(id)
+          .map(&.{{ through[1].id }}_count)
+          .sum
```

Comment:
This is definitely making more sql calls than before. Like I said above, we can't start at the opposite end of the association, we have to start from base and travel through the "throughs"/intermediary tables because those are the only names we know.

This is why I had to add the `def {{...}}_count` method to belongs to associations.